src/EditorFeatures/Core/Implementation/InlineRename/CommandHandlers/RenameCommandHandler.cs

```diff
@@ -77,17 +78,22 @@ private void HandlePossibleTypingCommand(CommandArgs args, Action nextHandler, A
                 // It's in a read-only area, so let's commit the rename and then let the character go
                 // through
 
-                CommitIfActiveAndCallNextHandler(args, nextHandler);
+                TryCommitIfActiveAndCallNextHandlerIfCommitted(args, nextHandler);
             }
         }
 
-        private void CommitIfActiveAndCallNextHandler(CommandArgs args, Action nextHandler)
+        private void TryCommitIfActiveAndCallNextHandlerIfCommitted(CommandArgs args, Action nextHandler)
         {
             if (_renameService.ActiveSession != null)
             {
                 var selection = args.TextView.Selection.VirtualSelectedSpans.First();
 
-                _renameService.ActiveSession.Commit();
+                if (!_renameService.ActiveSession.TryCommit())
+                {
+                    // If the active rename session was not committed, then do not allow the
+                    // command to propagate through.
```

Comment:
User experience question: should we just commit without a preview here? Should "preview" _only_ apply to explicitly committed rename sessions?